server/server.go

```diff
@@ -327,6 +327,23 @@ func (s *Server) Start(ctx context.Context) error {
 		return err
 	}
 	log.Tracef(ctx, "listening on port %s", s.ctx.Addr)
+	// TODO(tamird): remove this hack when we can update grpc
```

Comment:
Can you add a little bit of commentary? In particular, up to which point do we throw away connections and why is it fine to stop doing that at the end of this method.